geomesa-gt/geomesa-gt-partitioning/src/main/scala/org/locationtech/geomesa/gt/partition/postgis/dialect/procedures/PartitionMaintenance.scala

```diff
@@ -0,0 +1,58 @@
+/***********************************************************************
+ * Copyright (c) 2013-2022 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.gt.partition.postgis.dialect
+package procedures
+
+import scala.util.hashing.MurmurHash3
+
+/**
+ * High level procedure to manage data shuffling between write-ahead table and partitioned tables
+ */
+object PartitionMaintenance extends SqlProcedure with CronSchedule {
+
+  override def name(info: TypeInfo): String = s"${info.name}_partition_maintenance"
+
+  override def jobName(info: TypeInfo): String = s"${info.name}-part-maintenance"
+
+  override protected def createStatements(info: TypeInfo): Seq[String] =
+    Seq(proc(info)) ++ super.createStatements(info)
+
+  override protected def schedule(info: TypeInfo): String = {
+    val minute = info.partitions.cronMinute.getOrElse {
+      // spread out the cron schedule so that all the feature types don't run at the exact same time
+      // also don't run at same minute as roll-write-ahead (i.e. use 0-8)
+      // TODO this will mean data left in the write ahead table longer -
```

Comment:
Are the WA tables renamed to something?  If so, can the "new" WA table just be named what it will be named and then writes are directed to whatever table they belong to?

Reply:
to keep it simple the one that receives writes is always suffixed with `_writes` and then renamed during the roll, according to a seq. So you'll start with `_writes`, then `_000`, `_001`, etc. we could adopt the final name up front, but then we'd have to update the trigger when we roll tables, as well as track which table is receiving the writes so that we don't move the data out of it yet. the associated indices (e.g. `spatial_geom_000`) start out with the final names so that we don't have to rename them all. i guess it's pretty much equivalent either way.